Extensions/mass_plus.js

```diff
@@ -227,60 +166,42 @@ XKit.extensions.mass_plus = new Object({
-
-		this.search_url = "http://" + m_search_url + ".tumblr.com/archive";
-
+		var blog_shortname = document.location.href.substring(document.location.href.indexOf('mega-editor/') + 12);
+		if (blog_shortname .indexOf("/") !== -1) { blog_shortname = blog_shortname.substring(0, blog_shortname.indexOf("/")); }
+		if (blog_shortname .indexOf("?") !== -1) { blog_shortname = blog_shortname.substring(0, blog_shortname.indexOf("?")); }
+		if (blog_shortname .indexOf("#") !== -1) { blog_shortname = blog_shortname.substring(0, blog_shortname.indexOf("#")); }
 		this.search_found_count = 0;
 		this.search_last_timestamp = last_timestamp;
 		this.search_page = 0;
 		this.search_found_posts = [];
+		this.search_url = "http://api.tumblr.com/v2/blog/" + blog_shortname + ".tumblr.com/posts?api_key=" + XKit.api_key + "&tag=" + tag.toLowerCase();
 		this.search_next_page(tag.toLowerCase());
-
 	},
-
 	search_next_page: function(tag) {
-
-		var m_url = this.search_url + "?before_time=" + this.search_last_timestamp;
-		XKit.console.add("Fetching " + m_url);
+		var search_url = this.search_url + "&offset=" + (this.search_page * 20);
+		XKit.console.add("Fetching " + search_url);
 		GM_xmlhttpRequest({
 			method: "GET",
-			url: m_url,
+			url: search_url,
 			onerror: function(response) {
 				XKit.window.show("Can't fetch archive.","Could not fetch required pages to complete your request. Please try again later.","error","<div class=\"xkit-button default\" id=\"xkit-close-message\">OK</div>");
 			},
 			onload: function(response) {
-				$(".post", response.responseText).each(function() {
-					var m_tags = $(this).find(".tags").html();
-					if (typeof m_tags !== "undefined") {
-						m_tags = $.trim(m_tags);
-						var r_tags = m_tags.split("#");
-
-						for (var i=0;i<r_tags.length;i++) {
-
-							if (r_tags[i] === "") { continue; }
-
-							var mr_tag = r_tags[i].substring(0, r_tags[i].length - 1);
-							if (mr_tag.toLowerCase() === tag || r_tags[i].toLowerCase() === tag.toLowerCase()) {
-								var m_id = $(this).attr('data-post-id');
-								XKit.extensions.mass_plus.search_found_posts.push(m_id);
-								XKit.extensions.mass_plus.search_found_count++;
-							}
-						}
-					}
+				var response_object = JSON.parse(response.response);
+				var posts_array = response_object.response.posts;
+				if (!posts_array || posts_array.length === 0) {
+					XKit.extensions.mass_plus.search_results(tag);
+					return;
+				}
+				posts_array.forEach(function(post) {
+					XKit.extensions.mass_plus.search_found_posts.push(post.id);
+					XKit.extensions.mass_plus.search_found_count++;
 				});
 				if (XKit.extensions.mass_plus.search_found_count < 100 && XKit.extensions.mass_plus.search_page <= 5) {
-					var next_link = $("#next_page_link", response.responseText).attr('href');
-					if (typeof next_link === "undefined") {
+					if (posts_array.length < 20) {
```

Comment:
What is the significant of the number 20? Is it the default number of posts to be returned from the [`/posts`](https://www.tumblr.com/docs/en/api/v2#posts) API call? Regardless, we should be clear about what this number (and the number on line 181, assuming they're for the same purpose) represents.